### DISCUSSION TITLE
Using CSV Dict Writer with data package descriptor

### DISCUSSION THREAD

**Discussion by nirabpudasaini** [frictionlessdata/frictionlessdata.io#722] [Ideas]
You can use the data package descriptor with CSV Dict writer to define the output csv file that you want. In one of our internal key metric dash boarding project we use a utility function to achieve this. The utility function can be found in the following code snippet. This supports write and append mode. 

```python
import csv
from datapackage import Package


def csvDictWriterForDataPackageResource(dp, resource_name, write_format):
    # dp is path to datapackage
    # name of the resource
    # write_formart is insert or append - w for rewriting file, a for Appending
    package = Package(dp)
    resource = package.get_resource(resource_name)
    fieldnames = resource.schema.field_names
    output_path = "data/" + resource.descriptor["path"]

    outstream = csv.DictWriter(open(output_path, write_format, newline=""), fieldnames)
    return outstream
```

**Comment by rufuspollock** [comment]
@nirabpudasaini great! And could we add upsert 😉 and get this in a gist?

And i think you could pretty much skip using `Package` here and go plain - it does not really get you anything and reduces your dependencies.

**Comment by lwinfree** [comment]
Hi @nirabpudasaini! I'm working on cleaning up the issues in this repo, and I'm wondering if there is an action you'd like to take here (or an action you'd like someone else to take?) Thanks!

**Comment by roll** [comment]
I think it perfectly matches the "Ideas" section in Discussions
